src/compile/data/bin.ts

```diff
@@ -47,9 +45,20 @@ interface BinComponent {
 export class BinNode extends DataFlowNode implements NewFieldNode, DependentNode {
   private bins: Dict<BinComponent>;
 
+  public clone(): this {
+    const cloneObj = new (<any>this.constructor);
+    cloneObj.bins = duplicate(this.bins);
+    return cloneObj;
+  }
+
   constructor(model: Model) {
     super();
 
+    if (!model) {
```

Comment:
See [this comment](https://github.com/vega/vega-lite/pull/2086#pullrequestreview-32722315) -- we should get rid of this hack!